### ISSUE TITLE
gandi: updating deletes all record of same rtype

### ISSUE THREAD

**Issue by tristan-weil** [AnalogJ/lexicon#383]
When using Gandi provider, if you update a specific record with a specific type, all records with a matching name will be deleted.

The if-condition handling the case is not correct.

**Comment by adferrand** [comment]
Hello @tristan-weil. 

Just to confirm, are you saying that if we have for instance, a TXT and a A records for the same name, and you ask the TXT record to be deleted by its name, the two records are deleted?

**Comment by tristan-weil** [comment]
Hello,

Worse, the other records are deleted because it goes in the wrong part of the if-condition (line 180 in gandi.py or check the pr).

And indeed, I should have provided an example. Here it is:
a) without fix:
```
> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com TXT | grep srv3

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com LOC | grep srv3

> lexicon gandi --auth-token=xxx --api-protocol=rest create mydomain.com TXT --content "random text" --name srv3
RESULT
------
True

> lexicon gandi --auth-token=xxx --api-protocol=rest create mydomain.com LOC --content "43 21 43.528 N 71 05 06.284 W 12m" --name srv3 
RESULT
------
True

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com TXT | grep srv3
srv3 TXT  srv3.mydomain.com random text                                 3600 

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com LOC | grep srv3
srv3 LOC  srv3.mydomain.com 43 21 43.528 N 71 05 06.284 W 12m 3600

> lexicon gandi --auth-token=xxx --api-protocol=rest update mydomain.com LOC --content "44 21 43.528 N 71 05 06.284 W 12m" --name srv3 
RESULT
------
True

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com TXT | grep srv3

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com LOC | grep srv3
srv3 LOC  srv3.mydomain.com 44 21 43.528 N 71 05 06.284 W 12m 10800
```


b) with fix:
```
> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com TXT | grep srv3

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com LOC | grep srv3

> lexicon gandi --auth-token=xxx --api-protocol=rest create mydomain.com TXT --content "more random text" --name srv3
RESULT
------
True

> lexicon gandi --auth-token=xxx --api-protocol=rest create mydomain.com LOC --content "53 21 43.528 N 71 05 06.284 W 12m" --name srv3 
RESULT
------
True

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com TXT | grep srv3
srv3 TXT  srv3.mydomain.com more random text                            3600 

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com LOC | grep srv3
srv3 LOC  srv3.mydomain.com 53 21 43.528 N 71 05 06.284 W 12m 3600

> lexicon gandi --auth-token=xxx --api-protocol=rest update mydomain.com LOC --content "54 21 43.528 N 71 05 06.284 W 12m" --name srv3 
RESULT
------
True

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com TXT | grep srv3
srv3 TXT  srv3.mydomain.com more random text                            3600 

> lexicon gandi --auth-token=xxx --api-protocol=rest list mydomain.com LOC | grep srv3
srv3 LOC  srv3.mydomain.com 54 21 43.528 N 71 05 06.284 W 12m 10800
```

**Comment by adferrand** [comment]
Aouch. Since I wrote the RPC part of provider, not the Live REST API, I do not have appropriate account to refresh the records for integration tests. Since you have them, could you run the tests and update them in your PR ? I will merge and publish a fix right after.

To refresh records, you can follow the `Test recordings` section in https://github.com/AnalogJ/lexicon/blob/master/CONTRIBUTING.md. Do not forget to delete the old records from `tests/fixtures/cassettes/gandi/REST-IntegrationTests` before.

**Comment by tristan-weil** [comment]
looks good?

**Comment by adferrand** [comment]
Yes, just merge your PR!

**Comment by adferrand** [comment]
Released 3.1.7 with this fix included.
